tidb-lightning/tidb-lightning-data-source.md

```diff
@@ -0,0 +1,326 @@
+    ```
+
+#### Non-configurable options
+
+TiDB Lightning does not support every option supported by the `LOAD DATA` statement. For example:
+
+* There cannot be line prefixes (`LINES STARTING BY`).
+* The header cannot be simply skipped (`IGNORE n LINES`). The header must be valid column names.
+
+### Strict format
+
+TiDB Lightning works the best when the input files have a uniform size around 256 MiB. When the input is a single huge CSV file, TiDB Lightning can only process the file in one thread, which slows down import speed.
+
+This can be fixed by splitting the CSV into multiple files first. For the generic CSV format, there is no way to quickly identify where a row starts or ends without reading the whole file. Therefore, TiDB Lightning by default does *not* automatically split a CSV file. However, if you are certain that the CSV input adheres to certain restrictions, you can enable the `strict-format` setting to allow TiDB Lightning to split the file into multiple 256 MiB-sized chunks for parallel processing.
+
+```toml
+[mydumper]
+strict-format = true
+```
+
+In a strict CSV file, every field occupies only a single line. In other words, one of the following must be true:
+
+* Delimiter is empty.
+* Every field does not contain the terminator itself. In the default configuration, this means every field does not contain CR (`\r`) or LF (`\n`).
+
+If a CSV file is not strict, but `strict-format` is wrongly set to `true`, a field spanning multiple lines may be cut in half into two chunks, causing parse failure, or even quietly importing corrupted data.
+
+### Common configuration examples
+
+#### CSV
+
+The default setting is already tuned for CSV following RFC 4180.
+
+```toml
+[mydumper.csv]
+separator = ',' # If the data might contain comma (','), it is recommended to use '|+|' or other uncommon character combinations as separator.
+delimiter = '"'
+header = true
+not-null = false
+null = '\N'
+backslash-escape = true
+```
+
+Example content:
+
+```
+ID,Region,Count
+1,"East",32
+2,"South",\N
+3,"West",10
+4,"North",39
+```
+
+#### TSV
+
+```toml
+[mydumper.csv]
+separator = "\t"
+delimiter = ''
+header = true
+not-null = false
+null = 'NULL'
+backslash-escape = false
+```
+
+Example content:
+
+```
+ID    Region    Count
+1     East      32
+2     South     NULL
+3     West      10
+4     North     39
+```
+
+#### TPC-H DBGEN
+
+```toml
+[mydumper.csv]
+separator = '|'
+delimiter = ''
+terminator = "|\n"
+header = false
+not-null = true
+backslash-escape = false
+```
+
+Example content:
+
+```
+1|East|32|
+2|South|0|
+3|West|10|
+4|North|39|
+```
+
+## SQL
+
+When TiDB Lightning processes a SQL file, because TiDB Lightning cannot quickly split a single SQL file, it cannot improve the import speed of a single file by increasing concurrency. Therefore, when you import data from SQL files, avoid a single huge SQL file. TiDB Lightning works the best when the input files have a uniform size around 256 MiB.
```

Comment:
```suggestion
When TiDB Lightning processes a SQL file, because TiDB Lightning cannot quickly split a single SQL file, it cannot improve the import speed of a single file by increasing concurrency. Therefore, when you import data from SQL files, avoid a single huge SQL file. TiDB Lightning works best when the input files have a uniform size of around 256 MiB.
```